BinScatter/R/BinScatter.R

```diff
@@ -0,0 +1,258 @@
+  ## Define local functions
+  ##------------------------------------------------------------------------------------------------
+
+  ## Get midpoint of an interval (bin center for `binType` == "uniform")
+  midpoint <- function(interval) {
+    
+    interval <- as.character(interval) %>% str_sub(2,-2)
+    start    <- str_split(interval, ", ")[[1]][1] %>% as.numeric()
+    end      <- str_split(interval, ", ")[[1]][2] %>% as.numeric()
+    midpoint <- 0.5 * ( start + end )
+    
+    return(midpoint)
+  }
+  
+  ## Get mean of `x_var` in an interval (bin center for `binType` == "quantile")
+  meanval <- function(interval, vals) {
+    
+    interval <- as.character(interval) %>% str_sub(2,-2)
+    start    <- str_split(interval, ", ")[[1]][1] %>% as.numeric()
+    end      <- str_split(interval, ", ")[[1]][2] %>% as.numeric()
+    meanval  <- vals[start <= vals & vals < end] %>% mean()
+    
+    return(meanval)
+  }
+  
+  ## Create bin indicators and bin centers for `x_var`
+  make_bin_indicator <- function(var, data, nBins, binType, intercept, is_xvar = FALSE) {
+    
+    if (binType == "uniform") {
+      tempBinType <- "explicit"
+    } else if (binType == "quantile") {
+      tempBinType <- "quantile"
+    } else {
+      stop(sprintf("binType for %s must be uniform or quantile", var))
+    }
+    
+    # Create a data frame of bin indicators
+    vals           <- data[, eval(var)] %>% as.numeric()
+    bins           <- bin_data(vals, bins = nBins, binType = tempBinType) %>% factor(ordered = FALSE)
+    df_bins        <- data.table(bins) %>% one_hot()
+    names(df_bins) <- paste0(var, "_bin_", c(1:nBins))
+    
+    if (intercept) {
+      df_bins        <- df_bins[, 2:nBins]
+      names(df_bins) <- paste0(var, "_bin_", c(2:nBins))
+    }
+    
+    # Return additionally bin centers for the `x_var`
+    if (is_xvar) {
+      if (binType == "uniform") {
+        bin_center <- sapply(unique(bins), midpoint) %>% sort()
+      } else if (binType == "quantile") {
+        bin_center <- sapply(unique(bins), meanval, vals) %>% sort()
+      }
+      return(list(indicator = df_bins, center = bin_center))
+    } else {
+      return(df_bins)
+    }
+  }
+
+  
+  ##------------------------------------------------------------------------------------------------
+  ## Prepare data for regression
+  ##------------------------------------------------------------------------------------------------
+  
+  ## Clean data
+  data <- data %>% as.data.frame()
+  data <- data[, all_of(c(y_var, x_var, linpartial_var, binpartial_var))]
+  nRaw <- nrow(data)
+  data <- data %>% drop_na() 
+  nNA  <- nRaw - nrow(data)
+  if (nNA > 0) {
+    if (dropNA) { 
+      warning(sprintf("%s observations dropped due to missing value.", nNA)) 
+    } else {
+      stop("Missing values present. Set dropNA == TRUE to overide")
+    }
+  }
+  
+  df_reg <- data %>% select(eval(y_var))
```

Comment:
```suggestion
  df_reg <- data %>% dplyr::select(eval(y_var))
```

When I run the example code, I get the error `Error in select(., eval(y_var)) : unused argument (eval(y_var))` because there is a naming conflict for `select` from `dplyr` and `MASS`.